### DISCUSSION TITLE
it's better to have skywalking arm64 docker image

### DISCUSSION THREAD

**Discussion by mis98zb** [apache/skywalking#7590] [Q&A]
Please answer these questions before submitting your issue.

- Why do you submit this issue?
- [ ] Question or discussion
- [ ] Bug
- [*] Requirement
- [ ] Feature or performance improvement

___
### Requirement or improvement

With the arm64 support of Elasticsearch, the only thing prevents me to run skywalking in arm environments is just the lack of skywalking arm64 docker images.
May I know is there any plan about this?

thanks & best regards.

**Comment by wu-sheng** [comment]
Hi, the current committer team seems having no one working on this. If you want, you could open a pull request to add for dev or release. Once we have the docker file, we could do that in the release process.

**Comment by mis98zb** [comment]
hi, may I know where is skywalking built on?
Is it built on GitHub, or some customized environment?

For your information, we cross build docker images for both amd64 & arm64 using docker buildx, which needs some special settings in the build environment.
So maybe skywalking also needs some changes in the build environment…

**Reply by wu-sheng** [reply]
I can't read what you said. Please log in GitHub and update.

**Reply by mis98zb** [reply]
实在读不懂我就用中文写吧。
英文太菜了。

编译arm64版的docker image不是只提交一个docker file就完了的。
在x86上编译arm64的docker image需要用到docker的实验性功能，用buildx来编译。
这需要对编译的环境做一些设置。

所以我想问一下，现在skywalking是在哪儿编译的？github上编译的？还是有自己的可配置的编译环境？
然后再看看怎么支持arm64的编译。

**Reply by wu-sheng** [reply]
I didn't mean your English, I was confused by your previous content format. It was from mail, and the format is chaos.
Please keep all things in English, we need to make the process readable for global users.

For dev version, our docker image  built through this, https://github.com/apache/skywalking/blob/master/.github/workflows/publish-docker.yaml
In the official release, the docker image is from here. https://github.com/apache/skywalking-docker

**Reply by mis98zb** [reply]
oh, cannot see my previous reply...

This repository uses github workflow, I'm trying to understand it...

**Reply by wu-sheng** [reply]
Yes, all of our current processes are on the GitHub CI now. Because otherwise, we can't ask every committer pushing the images by theirselves and on local env.

**Reply by mis98zb** [reply]
The docker base image aop & ui used is apline version.
However, the apline version of openjdk does not support arm architecture.
Is it possible to change the base image to openjdk:11-jre?

**Reply by wu-sheng** [reply]
From tech perspective, you can, but AFAIK there are more changes are required. Such as OpenSSL is required to install.

**Reply by mis98zb** [reply]
ok.
openjdk:11-jre contains all needed packages such as openssl & bash & gnupg & ca-certificates, so I have remove all apk instructions in docker file, :P
Build succeeded in github.
Since I don't know where to push the docker image, so I did not add --push parameter for buildx command.
So as I did not test the docker image.

How should I do in the next step? Create a pull request directly?

btw, thanks for the network issue, it's so hard to submit code to github, and open the web page to check the pipeline result...

**Reply by wu-sheng** [reply]
If you want to update, there are 2 steps at least. Add one pull request to update base image, and all e2e checked.
Then whether GitHub Action provides ARM env already? If so, add that env to existing tests.
